src/modules/class-task-list.js

```diff
@@ -0,0 +1,76 @@
+import Task from './class-task.js';
+import isStorageAvailable from './local-storage-checker.js';
+
+const storageName = 'taskList';
+
+export default class TaskList {
+  constructor() {
+    this.data = [];
+    this.id = 1;
+    this.initUpdate();
+  }
+
+  initUpdate() {
+    if (isStorageAvailable('localStorage')) {
+      const data = JSON.parse(localStorage.getItem(storageName));
+
+      if (data && data.length !== 0) {
+        this.data = JSON.parse(localStorage.getItem(storageName));
+
+        const lastItem = this.data.reduce((prev, current) => {
+          const val = prev.id > current.id ? prev : current;
+          return val;
+        });
+        this.id = lastItem.id + 1;
+      }
+    }
+  }
+
+  addTask(task) {
+    if (isStorageAvailable) {
+      const taskObj = new Task(task, this.id);
+
+      this.data.push(taskObj);
+      localStorage.setItem(storageName, JSON.stringify(this.data));
+
+      this.id += 1;
+    }
+  }
+
+  removeTask(taskID) {
+    this.data = this.data.filter((i) => i.id !== taskID);
+    this.resetIds();
+    localStorage.setItem(storageName, JSON.stringify(this.data));
+  }
+
+  updateStatus(taskID, status) {
+    this.data = this.data.map((obj) => {
+      if (obj.id === taskID) {
+        return { ...obj, completed: status };
+      }
+
+      return obj;
+    });
+    localStorage.setItem(storageName, JSON.stringify(this.data));
+  }
+
+  renameTask(taskID, newTask) {
+    this.data = this.data.map((obj) => {
+      if (obj.id === taskID) {
+        return { ...obj, description: newTask };
+      }
+
+      return obj;
+    });
+    localStorage.setItem(storageName, JSON.stringify(this.data));
```

Comment:
- [ ] you should also save the completed status when an item is completed and you load the page it becomes `completed: false ` it should stay true, so please try to fix that